runtime/compiler/runtime/MetaData.cpp

```diff
@@ -1472,12 +1488,19 @@ createMethodMetaData(
    data->registerSaveDescription = comp->cg()->getRegisterSaveDescription();
 
 #if defined(J9VM_INTERP_AOT_COMPILE_SUPPORT)
-   if (vm->isAOT_DEPRECATED_DO_NOT_USE())
+   if (vm->isAOT_DEPRECATED_DO_NOT_USE()
+#if defined(JITSERVER_SUPPORT)
+      || comp->getPersistentInfo()->getRemoteCompilationMode() == JITServer::SERVER
```

Comment:
Since we have the comp object, a better test could be comp->isOutProcessCompilation()